source/common/upstream/health_checker_impl.cc

```diff
@@ -256,6 +260,16 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::decodeHeaders(
   }
 }
 
+void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::decodeData(Buffer::Instance& data,
+                                                                     bool end_stream) {
+  if (!parent_.expected_response_.empty()) {
+    absl::StrAppend(&body_, data.toString());
```

Comment:
We should accumulate up to 1Kb of data only here to avoid keeping large amounts of data if health response is large. Also you can make the `body_` to be a `Buffer` too, and move slices from `data` into `body_` to avoid copying. When you check for presence of the string you will need to call `linearize` on the `body_` first.

Reply:
Good point. Done! Thank you!